src/main/java/org/pac4j/jax/rs/pac4j/JaxRsProfileManager.java

```diff
@@ -0,0 +1,155 @@
+ * @since 1.0.0
+ *
+ */
+public class JaxRsProfileManager extends ProfileManager<CommonProfile> {
+
+    public JaxRsProfileManager(WebContext context) {
+        super(context);
+
+        ContainerRequestContext requestContext = getJaxRsContext().getRequestContext();
+        SecurityContext original = requestContext.getSecurityContext();
+        if (!(original instanceof Pac4JSecurityContext)) {
+            requestContext.setSecurityContext(new Pac4JSecurityContext(original));
+        }
+    }
+
+    protected JaxRsContext getJaxRsContext() {
+        return (JaxRsContext) this.context;
+    }
+
+    @Override
+    protected LinkedHashMap<String, CommonProfile> retrieveAll(boolean readFromSession) {
+        LinkedHashMap<String, CommonProfile> profiles = super.retrieveAll(readFromSession);
+
+        SecurityContext securityContext = getJaxRsContext().getRequestContext().getSecurityContext();
+        if (securityContext instanceof Pac4JSecurityContext) {
+            ((Pac4JSecurityContext) securityContext).setPrincipal(profiles);
+        }
+
+        return profiles;
+    }
+
+    @Override
+    public void logout() {
+        super.logout();
+
+        SecurityContext securityContext = getJaxRsContext().getRequestContext().getSecurityContext();
+        if (securityContext instanceof Pac4JSecurityContext) {
+            ((Pac4JSecurityContext) securityContext).unsetPrincipal();
+        }
+    }
+
+    private static class Pac4JSecurityContext implements SecurityContext {
+
+        private final SecurityContext original;
+
+        private Principal principal;
+        private Set<String> roles;
+
+        public Pac4JSecurityContext(SecurityContext original) {
+            this.original = original;
+        }
+
+        public void setPrincipal(LinkedHashMap<String, CommonProfile> profiles) {
+            if (profiles != null && !profiles.isEmpty()) {
+                final CommonProfile profile = ProfileHelper.flatIntoOneProfile(profiles).get();
+                principal = new PrincipalImpl(profile.getId());
+                roles = new HashSet<>();
+                profiles.values().stream().forEach(p -> roles.addAll(p.getRoles()));
+            }
+        }
+
+        public void unsetPrincipal() {
+            this.principal = null;
+            this.roles = null;
+        }
+
+        @Override
+        public Principal getUserPrincipal() {
+            if (principal != null) {
+                return principal;
+            } else {
+                return original != null ? original.getUserPrincipal() : null;
+            }
+        }
+
+        @Override
+        public boolean isUserInRole(String role) {
+            if (roles != null) {
+                return roles.contains(role);
+            } else {
+                return original != null && original.isUserInRole(role);
+            }
+        }
+
+        @Override
+        public boolean isSecure() {
+            return original != null && original.isSecure();
```

Comment:
`isSecure()` is not dependent on pac4j, only on the way the request was made (with https or not), so the JAX-RS runtime default `SecurityContext` already have this information.